### DISCUSSION TITLE
可以用ravu-zoom-r2-chroma.hook替代cscale(chroma upscale)功能吗

### DISCUSSION THREAD

**Discussion by yeezylife** [hooke007/mpv_PlayKit#12] [Q&A 问答] [answered]
` glsl-shaders              = "~~/shaders/ravu-zoom-r3.hook"`
` glsl-shaders-append  = "~~/shaders/SSimDownscaler.glsl"`
` glsl-shaders-append  = "~~/shaders/adaptive-sharpen.glsl"`
` glsl-shaders-append  = "~~/shaders/ravu-zoom-r2-chroma.hook"`

着色器配置我是这么写的,加上ravu-zoom-r2-chroma.hook这行就可以了吗..
我看有ravu-zoom-r2-chroma/ravu-zoom-r2-rgb/ravu-zoom-r2-yuv 三个不同的变体,不知道rgb/yuv文件是干嘛的..查了半天资料我已经麻了..

**Comment by hooke007** [comment, marked as answer]
-yuv 和 -rgb 都是作用在通道合并之后再全体upscale用的（手册01里有写）
-chroma 可以代替cscale用，算是 krig 的下位代替品

（除了你这几个画的斜线啥意思我没看懂外，你的路径写错了）
`glsl-shaders-append = "~~/shaders/ravu-zoom-r2-chroma.hook"`

**Comment by yeezylife** [comment]
> -yuv 和 -rgb 都是作用在通道合并之后再全体upscale用的（手册01里有写）
> -chroma 可以代替cscale用，算是 krig 的下位代替品
> 
> （你的路径写错了）
> `glsl-shaders-append = "~~/shaders/ravu-zoom-r2-chroma.hook"`

感谢回复,我复制过来就变成中间带横杠了..
因为chroma upscale感知度不高,所以我就想选ravu-zoom-r2-chroma这种相对消耗不上不下的..我再去读一读 手册01

**Comment by hooke007** [comment]
>因为chroma upscale感知度不高

除了这种特例以外，cscale绝大多数情况都无所谓的，默认的bilinear即可
https://github.com/hooke007/MPV_lazy/tree/master/Temp#cscale%E8%89%B2%E5%BA%A6%E5%8D%87%E9%A2%91%E7%AE%97%E6%B3%95%E7%9A%84%E5%AF%B9%E6%AF%94

-yuv -rgb 和 不带后缀的区别，看图（前两个在combing planes通道合并阶段后，一个在合并阶段前）
此外ravu整个系列作为预处理的优先层级非常高，就算你配置里把它位置写在别的着色器之后也会强制被先执行

![Snipaste_2021-08-07_20-01-09](https://user-images.githubusercontent.com/41094733/128599557-80385300-db9a-4bdc-ad7f-34d0fe55825d.jpg)
![Snipaste_2021-08-07_20-01-38](https://user-images.githubusercontent.com/41094733/128599561-7276e3f4-d41b-4482-94f9-3f01c73be1a7.jpg)

**Comment by yeezylife** [comment]
![11](https://user-images.githubusercontent.com/30175425/128599848-73635120-6e5e-4091-aa87-be7415e28675.png)

确实,我的配置和图二一样,是在合并阶段前

**Comment by hooke007** [comment]
而且你用zoom这个变体的话，就没内置算法的事情了，相应的两个SSim也没他们的事情了

**Comment by yeezylife** [comment]
> 而且你用zoom这个变体的话，就没内置算法的事情了，相应的两个SSim也没他们的事情了

![2222](https://user-images.githubusercontent.com/30175425/128600311-6b54afda-15e7-4a9a-b11e-cdb260055010.png)
![11](https://user-images.githubusercontent.com/30175425/128600316-8165c70f-9a9c-4bc8-b416-f177b90a9ed7.png)
是指ravu-zoom-r2-chroma.hook还是ravu-zoom-r3.hook?
我看截图里,SSimDownscaler.glsl和adaptive-sharpen.glsl都是正常被使用的啊
我用ravu-zoom-r3.hook是因为它能直接缩放到指定的分辨率 不会出现达不到最低放大比率不被调用的情况.

**Comment by hooke007** [comment]
看错了，zoom （不带chroma）不影响 dscale，应该说的是SSimSuperRes没用了

**Comment by yeezylife** [comment]
> 看错了，zoom （不带chroma）不影响 dscale，应该说的是SSimSuperRes没用了

嗯 不管是SSimSuperRes 还是自带的scale算法都不会工作,因为已经被zoom （不带chroma）放大到屏幕的分辨率了,接下来好像走了一次scale=bilinear>(adaptive-sharpen)之后就输出了
实际上我conf里的scale是你默认的ewa_lanczossharp

**Comment by hooke007** [comment]
>scale=bilinear

这一步是远古bug，但是不影响（画面）

**Comment by yeezylife** [comment]
cscale 已经用回catmull_rom了,cscale部分的消耗是ravu-zoom-r2-chroma/ewa_lanczossharp的四分之一到五分之一..效果反正我看不出区别.. 
据说catmull_rom=bicubic50,我想用bicubic60但mpv中没有相关的项目,不知道bicubic_fast对应的是多少.

**Reply by hooke007** [reply]
madvr中的cubicXX应该代表的是B值，计算目标可能是B+C=1

### softbicubic60
```
scale        = bcspline
scale-param1 = 0.6
scale-param2 = 0.4
```

**Reply by yeezylife** [reply]
我转了一圈又用回去krig了, 我发现krig明亮场景感知确实不强,但是在昏暗场景里的亮光,颜色差别特别大,肉眼可见的区别. (比如演唱会舞台上的灯光/布景,晚上的路灯,在昏暗房间看电视之类的场景) 
另外krig和ravu还原的颜色类似,细节处有不同,都能用,其他的普通算法在昏暗+亮光场景还原不出这种颜色.
另外其实spline16就和bicubic60/75类似了,我也懒得折腾了
